crates/sui-replay/src/data_fetcher.rs

```diff
@@ -210,6 +212,12 @@ impl DataFetcher for Fetchers {
             Fetchers::NodeStateDump(q) => q.get_epoch_change_events(reverse).await,
         }
     }
+    async fn get_chain_id(&self) -> Result<String, ReplayEngineError> {
+        match self {
+            Fetchers::Remote(q) => q.get_chain_id().await,
+            Fetchers::NodeStateDump(q) => q.get_chain_id().await,
```

Comment:
nit - since the handling is the same
```suggestion
            Fetchers::Remote(q) | Fetchers::NodeStateDump(q) => q.get_chain_id().await,
```

Reply:
That only works if q is same type in both cases